relayer/relayer/src/trader/mod.rs

```diff
@@ -0,0 +1,302 @@
+            ..
+        } = &*config;
+
+        let most_azero = MostInstance::new(
+            azero_contract_address,
+            azero_contract_metadata,
+            *azero_ref_time_limit,
+            *azero_proof_size_limit,
+        )?;
+
+        let router = RouterInstance::new(
+            &router_address
+                .clone()
+                .ok_or(TraderError::MissingRequired("router_address".to_owned()))?,
+            router_metadata,
+            *azero_ref_time_limit,
+            *azero_proof_size_limit,
+        )?;
+
+        let azero_ether_address =
+            azero_ether_address
+                .clone()
+                .ok_or(TraderError::MissingRequired(
+                    "azero_ether_address".to_owned(),
+                ))?;
+
+        let azero_ether = AzeroEtherInstance::new(
+            &azero_ether_address,
+            azero_ether_metadata,
+            *azero_ref_time_limit,
+            *azero_proof_size_limit,
+        )?;
+
+        let wrapped_azero_address =
+            AccountId::from_str(&azero_wrapped_azero_address.clone().ok_or(
+                TraderError::MissingRequired("azero_wrapped_azero_address".to_owned()),
+            )?)
+            .map_err(|err| TraderError::NotAccountId(err.to_owned()))?;
+
+        let wrapped_ether_address = eth_wrapped_ether_address
+            .clone()
+            .ok_or(TraderError::MissingRequired(
+                "eth_wrapped_ether_address".to_owned(),
+            ))?
+            .parse::<Address>()?;
+
+        let wrapped_ether = WETH9::new(wrapped_ether_address, eth_signed_connection.clone());
+
+        let whoami_azero = azero_signed_connection.account_id();
+        let whoami_eth = eth_signed_connection.address().to_string();
+
+        info!("Starting");
+
+        select! {
+            cb_event = circuit_breaker_receiver.recv() => {
+                warn!("Exiting due to a circuit breaker event {cb_event:?}.");
+                Ok(cb_event?)
+            },
+
+            _ = async {
+
+                loop {
+                    debug!("Ping");
+
+                    let azero_balance = azero_signed_connection
+                        .get_free_balance(whoami_azero.to_owned(), None)
+                        .await;
+
+                    info!("{whoami_azero} has a balance of: {azero_balance} Azero.");
+
+                    let current_base_fee = match most_azero.get_base_fee(azero_signed_connection.as_connection()).await {
+                        Ok(amount) => {
+                            info!("Current base fee: {amount} pA0");
+                            amount},
+                        Err(why) => {
+                            warn!("Query to `get_base_fee` has failed {why:?}.");
+                            continue;
+
+                        },
+                    };
+
+                    // check azero balance
+                    if azero_balance > current_base_fee + ONE_AZERO  {
```

Comment:
To perform any swaps AZERO balance should be much higher than this - we not only want to get some ETH back on the Ethereum side, but we want the 20% buffer to be enough to cover all bridging fees + txn fees + slippage.

With the current condition we would run out of the funds pretty quickly